pupepat/fitting.py

```diff
@@ -135,81 +133,97 @@ def fit_cutout(data, source, plot_filename, image_filename, header, id, fit_circ
     brightness_guess = np.median(cutout[np.logical_and(r < outer_radius_guess, r > inner_radius_guess)])
     inner_brightness_guess = np.median(cutout[r < inner_radius_guess])
 
-    initial_model = elliptical_annulus(x0_inner=x0, y0_inner=y0,
-                                       x0_outer=x0, y0_outer=y0,
-                                       amplitude_inner=inner_brightness_guess, amplitude_outer=brightness_guess,
-                                       a_inner=inner_radius_guess, b_inner=inner_radius_guess,
-                                       a_outer=outer_radius_guess,
-                                       b_outer=outer_radius_guess,
-                                       background=np.median(data))
-
-    if not fit_gradient:
-        initial_model.x_slope.fixed = True
-        initial_model.y_slope.fixed = True
-
+    x, y = np.meshgrid(np.arange(cutout.shape[1]), np.arange(cutout.shape[0]))
+    error = np.sqrt(np.abs(cutout))
     if fit_circle:
-        initial_model.theta_inner.fixed = True
-        initial_model.theta_outer.fixed = True
-        initial_model.b_inner.tied = lambda x: x.a_inner
-        initial_model.b_outer.tied = lambda x: x.a_outer
+        best_fit_parameter_names = ['x0_inner', 'y0_inner', 'r_inner', 'amplitude_inner',
+                                    'x0_outer', 'y0_outer', 'r_outer', 'amplitude_outer', 'background']
+        best_fit_solution = minimize(lambda *args: -ln_likelihood_circle(*args),
+                                     [x0, y0, inner_radius_guess, inner_brightness_guess,
+                                      x0, y0, outer_radius_guess, brightness_guess, np.median(data)],
+                                     method='Nelder-Mead', args=(x, y, cutout, error),
```

Comment:
I'm having some luck with the 'Powell' minimization method. On the (limited) examples I tried, it has noticeably better performance than 'Nelder Mead' with equally good convergence, and furthermore it actually managed to find a doughnut on coj1m003-kb05-20180603-0070-e00.fits.fz.

Reply:
Testing now.

Reply:
Nice. Results look good.